nomt/src/store/mod.rs

```diff
@@ -60,12 +125,48 @@ impl Store {
     /// After this function returns, accessor methods such as [`Self::load_page`] will return the
     /// updated values.
     pub fn commit(&self, tx: Transaction) -> anyhow::Result<()> {
+        let mut sync = self.sync.lock();
+        sync.sync_seqn += 1;
+
+        // store the sequence number of this sync.
+        let sync_seqn = sync.sync_seqn;
+
         self.shared.values.commit(tx.batch);
-        self.shared.pages.commit(tx.new_pages)?;
+        let prev_wal_size = self.shared.pages.sync_begin(tx.new_pages, sync_seqn)?;
         if let Some(new_root) = tx.new_root {
             *self.shared.root.lock() = new_root;
         }
-        self.shared.values.sync();
+        let data = self.shared.values.prepare_sync();
```

Comment:
the inconsistency between `Beatree::prepare_sync` and `Bitbox::sync_begin`,  both in terminology and order, is slightly grating

Reply:
my bad. Not even sure if either of those describes it.